code/game/objects/items/weapons/grenades/station.dm

```diff
@@ -0,0 +1,144 @@
+/**
+	Makes an instant station
+**/
+
+/obj/item/weapon/grenade/station
+	name = "Syndicate pocket sat"
+	desc = "A pocket satellite for your nefarious deeds. Find a nice empty space to prime and set it down - don't be right under it when you trigger it."
+	var/range = 7
+	var/walltype = /turf/simulated/wall/syndicate
+	var/floortype = /turf/simulated/floor/dark
+	var/doortype = /obj/structure/airshield
+	var/soundpath = 'sound/effects/bumpinthenight.ogg'
+	var/insidedoor = null
+
+/obj/item/weapon/grenade/station/prime()
+	playsound(src, soundpath, 75, 1)
+	var/turf/source = get_turf(src)
+	if(!istype(source, get_base_turf(source.z)))
+		visible_message("<span class='danger'>\The [src] cancels its detonation!</span>")
+		active = FALSE
+		icon_state = initial(icon_state)
+		return
+	for(var/mob/living/M in view(source, range))
```

Comment:
shouldn't this be `range()`?

Reply:
range is actually a var on the grenade

Reply:
ok but I'm talking about
```suggestion
	for(var/mob/living/M in range(source, range))
```

Reply:
Also, you should probably get this list of turfs once and pass it to the various procs. I believe currently this would result in mobs being affected in a square range while the proc that generates the satellite acts on a circular range.